hs/rsh/stdlib.rsh

```diff
@@ -35,8 +35,10 @@ export const fromMaybe = (v, onNull, onSome) => {
   case None: return onNull();
   case Some: return onSome(v); } };
 
-export const isSome = m => fromMaybe(m, () => false, (_) => true);
-export const isNone = m => fromMaybe(m, () => true, (_) => false);
+// TODO - this definitely emits optimized teal code, but it no longer checks that the data is actually a proper Maybe type...
+// TODO - also optimize isLeft/isRight, etc
+export const isSome = m => _dataTag(m) === 1;
+export const isNone = m => _dataTag(m) === 0;
```

Comment:
Idea: The new thing is `DLE_DataTag` but the SL feature is `isDataVariant(variantName, otherVariants, instance)` and the type checking happens just in SL but compiles to DL